script.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    while i < length
+      temp << self[i] if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    while i < length
+      return true if yield(self[i]) && yield(self[i + 1])
+
+      i += 1
+      return false
+    end
+  end
+
+  def my_any?
+    i = 0
+    while i < length
+      return true if yield(self[i]) || yield(self[i + 1])
+
+      i += 1
+      return false
+    end
+  end
+
+  def my_none?
+    i = 0
+    while i < length
+      return true unless yield(self[i])
+
+      i += 1
+      return false
+    end
+  end
+
+  def my_count
+    cont = 0
+    i = 0
+    while i < length
+      cont += 1 if yield(self[i])
+      i += 1
+    end
+    cont
+  end
+
+  def my_map(my_proc = nil)
+    temp = []
+    i = 0
+    while i < length
+      temp << if my_proc
+                my_proc.call(self[i])
+              else
+                yield(self[i])
+              end
+      i += 1
+    end
+    temp
+  end
+
+  def my_inject
+    i = 1
+    temp = self[0]
+    while i < length
+      temp = yield(temp, self[i])
+      i += 1
+    end
+    temp
+  end
+end
```

Comment:
- For `my_inject` handle the following cases 
1. when a symbol is specified, it combines each element of the collection by applying the symbol as a named method

Built-in method
`Range.new(5, 10).inject(2, :*)` returns `302400`

Your method gives the following error

> ArgumentError:
  wrong number of arguments (given 2, expected 0)

2. when a binary operation is specified by a block, it should apply a binary operation, combining all elements of enum.

Built-in method
`(5..10).inject { |sum, n| sum + n }` returns `45`

Your method gives the following error

> ArgumentError:
        wrong number of arguments (given 1, expected 0)

Reply:
method updated